src/main/java/hudson/plugins/analysis/core/AbstractResultAction.java

```diff
@@ -104,12 +106,12 @@ public ToolTipProvider getToolTipProvider() {
      *
      * @return the associated build of this action
      */
-    public final AbstractBuild<?, ?> getOwner() {
+    public final Run<?, ?> getOwner() {
```

Comment:
I've read the document: "Change type of API field --> Breaks compatibility" obviously. I think we have to keep backward compatibility when we **are able** to do it. I think in this particular case it's not possible.

Anyway, I've tested the last release of Checkstyle Plugin and it works fine, since the main extension point (```perform``` method in Publisher and constructors) are backward compatible.